access-common/src/test/java/edu/unc/lib/dl/ui/service/NeighborQueryTest.java

```diff
@@ -0,0 +1,317 @@
+		
+		assertEquals(WINDOW_SIZE, results.size());
+		int indexOfTarget = indexOf(results, targetMd);
+		assertEquals(WINDOW_SIZE - 3, indexOfTarget);
+		assertPrecedingResults(results, indexOfTarget);
+		assertSucceedingResults(results, indexOfTarget);
+	}
+	
+	@Test
+	public void testNeighborsBigTest() throws Exception {
+		populateNeighborhood(ResourceType.File, 500, 500, ResourceType.File,
+				rootPid, collectionPid, folderPid);
+		
+		BriefObjectMetadataBean targetMd = getMetadata(targetPid);
+		
+		long start = System.nanoTime();
+		List<BriefObjectMetadataBean> results = queryLayer.getNeighboringItems(targetMd, WINDOW_SIZE, groups);
+		log.info("Executed neighbors query in {}", (System.nanoTime() - start));
+		
+		assertEquals(WINDOW_SIZE, results.size());
+		int indexOfTarget = indexOf(results, targetMd);
+		assertEquals(WINDOW_SIZE / 2, indexOfTarget);
+		assertPrecedingResults(results, indexOfTarget);
+		assertSucceedingResults(results, indexOfTarget);
+	}
+	
+	private void populateNeighborhood(ResourceType targetType, int preceding, int succeeding, 
+			ResourceType neighborType, PID... ancestors) throws Exception {
+		List<SolrInputDocument> docs = new ArrayList<>();
+		
+		addNeighbors(docs, preceding, PRECEDING_PREFIX, neighborType, ancestors);
+		
+		addObject(docs, targetPid, "middle", targetType, ancestors);
+		
+		addNeighbors(docs, succeeding, SUCCEEDING_PREFIX, neighborType, ancestors);
+		
+		server.add(docs);
+		server.commit();
+	}
+	
+	private void addNeighbors(List<SolrInputDocument> docs, int count, String prefix, ResourceType type,
+			PID... ancestors) {
+		for (int i = 0; i < count; i++) {
+			PID neighborPid = makePid();
+			addObject(docs, neighborPid, prefix + i, type, ancestors);
+		}
+	}
+	
+	private void addObject(List<SolrInputDocument> docs, PID pid, String title, ResourceType type,
+			PID... ancestors) {
+		SolrInputDocument doc;
+		if (type.equals(ResourceType.File)) {
+			doc = makeFileDocument(pid, title, ancestors);
+		} else {
+			doc = makeContainerDocument(pid, title, type, ancestors);
+		}
+		addAclProperties(doc, "public", "admin");
+		docs.add(doc);
+	}
+	
+	private List<SolrInputDocument> populate() {
+		List<SolrInputDocument> docs = new ArrayList<SolrInputDocument>();
+
+		SolrInputDocument newDoc;
+		newDoc = makeContainerDocument(rootPid, "Root", ResourceType.Folder);
+		addAclProperties(newDoc, "public");
+		docs.add(newDoc);
+		
+		newDoc = makeContainerDocument(collectionPid, "Collection", ResourceType.Collection, rootPid);
+		addAclProperties(newDoc, "public", "admin");
+		docs.add(newDoc);
+		
+		newDoc = makeContainerDocument(folderPid, "Collection", ResourceType.Folder, rootPid, collectionPid);
+		addAclProperties(newDoc, "public", "admin");
+		docs.add(newDoc);
+
+		return docs;
+	}
+	
+	public SolrInputDocument makeContainerDocument(PID pid, String title, ResourceType type, PID... ancestors) {
```

Comment:
why is this method (and the four following) public?

Reply:
Oh, they were copy and pasted from a test utility in the fcrepo4 branch, i should change them to private here